### ISSUE TITLE
Euler-Lagrange equations derivation

### ISSUE THREAD

**Issue by marekyggdrasil** [allofphysicsgraph/graph#3]
Some key derivations that Euler-Lagrange derivation depends on

- Interchangeability of integration and differentiation (order of operations)
- Integration by parts
- Total differential (derived using chain rule)
- Fundamental Theorem of Calculus
- Fundamental Lemma of Calculus of Variations

Comment:

- Fundamental Lemma of Calculus of Variations has a proof complicated to represent on the graph, this will be addressed later in another issue.
- Need to add a new type of edge: `deduce`, which will be able to connect vertices based on a predicate, #1
- the `differential 2-form` is not supported by Sympy, we might need to use different computer algebra system or implement our own way to check correctness of that step, #2

**Comment by marekyggdrasil** [comment]
blocked by

https://github.com/allofphysicsgraph/graph/issues/7
https://github.com/allofphysicsgraph/graph/issues/1
